devdoc/buffer_requirements.md

```diff
@@ -39,7 +39,7 @@ extern BUFFER_HANDLE BUFFER_create(const unsigned char* source, size_t size);
 
 BUFFER_create creates a new buffer from the memory at source, having size "size".
 **SRS_BUFFER_02_001: [**If source is NULL then BUFFER_create shall return NULL.**]** 
-**SRS_BUFFER_02_002: [**Otherwise, BUFFER_create shall allocate memory to hold size bytes and shall copy from source size bytes into the newly allocated memory.**]** 
+**SRS_BUFFER_02_002: [**Otherwise, BUFFER_create shall allocate memory to hold size bytes and shall copy from source size bytes into the newly allocated memory. If size parameter is 0 then buffer field of BUFFER_TAG shall still hold a valid pointer whose content is '\0' and size field shall be 0.**]** 
```

Comment:
with regards to "That's because I think there is no reason to allocate anything, simply setting the internal buffer to NULL should be good", this is not going to work if internal buffer is NULL e.g. IoTHubMessage_CreateFromByteArray will fail. We want to make sure when size is 0 a valid pointer is returned and can be freed.